src/main/webapp/app/pages/curation/collapsible/MutationCollapsible.tsx

```diff
@@ -173,188 +179,216 @@ const MutationCollapsible = ({
+                          label="Level of Evidence in other solid tumor types"
+                          name="propagationLevel"
+                          options={[]} // Todo
+                        />
+                        <RealtimeDropdownInput
+                          fieldKey={`mutations/${firebaseIndex}/tumors/${tumorIndex}/TIs/${tiIndex}/treatments/${treatmentIndex}/propagationLiquid`}
+                          label="Level of Evidence in other liquid tumor types"
+                          name="propagationLiquidLevel"
+                          options={[]}
+                        />
+                        <RealtimeDropdownInput
+                          fieldKey={`mutations/${firebaseIndex}/tumors/${tumorIndex}/TIs/${tiIndex}/treatments/${treatmentIndex}/fdaLevel`}
+                          label="FDA Level of Evidence"
+                          name="propagationLiquidLevel"
+                          options={[]}
+                        />
+                        <RealtimeTextAreaInput
+                          fieldKey={`mutations/${firebaseIndex}/tumors/${tumorIndex}/TIs/${tiIndex}/treatments/${treatmentIndex}/description`}
+                          inputClass={styles.textarea}
+                          label="Description of Evidence"
+                          labelIcon={
+                            <GeneHistoryTooltip
+                              historyData={parsedHistoryList}
+                              location={`${CANCER_TYPE_THERAPY_INDENTIFIER}${getMutationName(mutation)}, ${cancerTypeName}, ${
+                                treatment.name
+                              }`}
+                            />
+                          }
+                          name="evidenceDescription"
+                        />
+                        <div className="mb-2">
+                          <AutoParseRefField summary={treatment.description} />
+                        </div>
+                      </Collapsible>
+                    );
+                  })
+                );
+              }, [])}
+            </Collapsible>
+            <ModifyCancerTypeModal
+              includedCancerTypes={tumor.cancerTypes}
+              excludedCancerTypes={tumor.excludedCancerTypes || []} // update model to include excluded cancer types
+              open={openCancerTypesUuid === tumor.cancerTypes_uuid}
+              onConfirm={async (includedCancerTypes, excludedCancerTypes) => {
+                const newTumor = _.cloneDeep(tumor);
+                newTumor.cancerTypes = includedCancerTypes;
+                newTumor.excludedCancerTypes = excludedCancerTypes;
+                if (!newTumor.excludedCancerTypes_uuid) {
+                  newTumor.excludedCancerTypes_uuid = generateUuid();
+                }
+                await updateTumor(cancerTypeFirebasePath, newTumor);
```

Comment:
@calvinlu3 what's our general strategy when firebase update fails for any reason?

Reply:
Also our strategy overall on transection when there are multiple edits across different firebase paths within same transection.